modules/storage/src/service/dispatch.rs

```diff
@@ -0,0 +1,69 @@
+use super::*;
+use bytes::Bytes;
+use futures::{Stream, TryStreamExt};
+use sha2::{digest::Output, Sha256};
+
+/// A common backend, dispatching to the ones we support.
+///
+/// This is required due to the "can't turn into object" problem, which we encounter for this trait
+/// (due to using async traits and function level type arguments). The only alternative would be
+/// to propagate the specific type up to the root level. However, that would also mean that actix
+/// handlers would be required to know about that full type to extract it as application
+/// data.
+///
+/// NOTE: Right now we only have one type (filesystem), but the goal is to have an additional one
+/// soon (e.g. S3)
+#[derive(Clone, Debug)]
+pub enum DispatchBackend {
+    Filesystem(FileSystemBackend),
+}
+
+impl StorageBackend for DispatchBackend {
```

Comment:
Ahhh, `DispatchBackend` is-a form of storage. Ignore previous.

Reply:
I reply to these in order. :smile: